src/cursor/common.rs

```diff
@@ -28,25 +30,26 @@ use crate::{
 #[derivative(Debug)]
 pub(super) struct GenericCursor<P, T>
 where
-    P: GetMoreProvider<DocumentType = T>,
+    P: GetMoreProvider,
 {
     #[derivative(Debug = "ignore")]
     provider: P,
     client: Client,
     info: CursorInformation,
-    buffer: VecDeque<T>,
+    buffer: VecDeque<RawDocumentBuf>,
```

Comment:
Since deserialization is no longer eager, we needed to update the stored buffer for the cursors to just contain raw BSON. Deserialization will happen as the cursor is iterated.

A cascading effect of this is that all of the cursor machinery (traits, result types, etc.) could have their generic parameters removed.